relayer-cli/src/commands/tx/client.rs

```diff
@@ -250,24 +259,36 @@ impl Runnable for TxUpgradeClientCmd {
         let client = ForeignClient::find(src_chain, dst_chain, &self.client_id)
             .unwrap_or_else(exit_with_unrecoverable_error);
 
-        // Assumption: this query is run while the chain is halted as a result of an upgrade
-        let src_upgrade_height = {
-            let src_application_height = match client.src_chain().query_latest_height() {
+        let mut src_application_latest_height = match client.src_chain().query_latest_height() {
+            Ok(height) => height,
+            Err(e) => Output::error(format!("{}", e)).exit(),
+        };
+
+        info!(
+            "Source application latest height: {}",
+            src_application_latest_height
+        );
+
+        // In order to perform the client upgrade, the chain is paused at the height specified by
+        // the user. When the chain is paused, the application height reports a height of 1 less
+        // than the height according to Tendermint. As a result, the target height at which the
+        // upgrade occurs at (the application height) is 1 less than the height specified by
+        // the user, hence the strictly less-than check in the while loop.
+        while src_application_latest_height < self.target_upgrade_height {
```

Comment:
How about we change this to 
```rust
while src_application_latest_height != self.target_upgrade_height - 1
```
It makes it more explicit what the desired state is for when we come out of this loop. Also, this currently infinite loops because `src_application_height` will always be smaller than `target_upgrade_height` (by at least 1)